CMakeLists.txt

```diff
@@ -16,13 +16,18 @@ project(
     "The 'build-all' instructions for all examples for the book CMake tips and tricks"
   LANGUAGES CXX)
 
-# Set build type to "Debug" if not set.
-if(NOT DEFINED CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE Debug)
-endif()
+# Set the default build type for single-config generators.
+# The build type variable is still overridable from outside.
+set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Default build type for single-config generators")
 
-# Print current build type
-message("Current build type ${CMAKE_BUILD_TYPE}")
+# Print generator type and build type (if applicable)
+get_property(is_multi_config_generator GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if (NOT is_multi_config_generator)
+  message(STATUS "Using a single-config generator (${CMAKE_GENERATOR})")
+  message(STATUS "Current build type is `${CMAKE_BUILD_TYPE}`")
+else()
+  message(STATUS "Using a multi-config generator (${CMAKE_GENERATOR})")
+endif ()
```

Comment:
I'm questioning the utility of this code to begin with. Is this a one-time thing that is relevant for showcasing differences between multi-config and single-config generators? If not, then any messages printed during configure time are just noise that should be printed with `VERBOSE` at least, but definitely not with `STATUS`.

Reply:
Printing such messages is useful for diagnosing the CI/CD pipeline issues and build environment issues. I also tend to print out the things like compiler flags, the toolchain version, and so on.

Reply:
CI is where the developer is in full control of the environment or at least they (ought to) have full knowledge of what software is on the CI machines. Even then, CMake allows the generator to be explicitly selected with the `-G` flag.

For regular CMake code that does nothing but describe build and usage requirements, the nature of the generator (or the config type) does not matter.

Reply:
The code above is for the book's "Changing build type" part. It allows user to get visual feedback when the build type changes, so even a newbie can confirm that the build type is "indeed" changed as intended. I tried to keep it simple in my first revision but changed it according to @alexreinking's feedbacks in issue #3 .

Reply:
Right. I wasn't aware of what this was all about. In that case however, you have to make sure readers on Windows use the `NMake Makefiles` generator, which ships by default with the VS toolchain.

Reply:
I already got a callout about this topic in 'Changing the build type' section. I also added a suggestion for using NMake for following the example in Windows as well.